helix-loader/Cargo.toml

```diff
@@ -30,7 +30,7 @@ log = "0.4"
 # cloning/compiling tree-sitter grammars
 cc = { version = "1" }
 threadpool = { version = "1.0" }
-tempfile = "3.9.0"
+tempfile = "3.10.0"
```

Comment:
The non-dev-dependency usage of tempfile in helix-loader seems odd to me:

https://github.com/helix-editor/helix/blob/5c567f31e236134c0e27dc689c8ad07ef5f9b5d1/helix-loader/src/grammar.rs#L447-L448

@pascalkuthe do we need the tempfile dependency to keep that data alive or can we just assign it? I'm not sure what the `TempPath::from_path` call is accomplishing

Reply:
`TempPath` is a drop guard that will ensure the path is deleted when it's dropped (so even if we panic or return early due to an error we don't want to keep object files lying around). You could write something like that yourself but we already depend on tempfile elsewhere (both as dev and non-dev dependency) so it seemed easier to just pull it im resther than recreating that